fairing/preprocessors/function.py

```diff
@@ -0,0 +1,43 @@
+import fairing
+import glob
+import os
+from pathlib import Path
+
+from fairing.constants import constants
+from .base import BasePreProcessor
+
+FUNCTION_SHIM = 'function_shim.py'
+
+
+class FunctionPreProcessor(BasePreProcessor):
+    def __init__(self,
+                 function_name,
+                 executable=None,
+                 path_prefix=constants.DEFAULT_DEST_PREFIX,
+                 output_map={}):
+
+        super().__init__(
+            output_map=output_map,
+            path_prefix=path_prefix)
+
+
+        fairing_dir = os.path.dirname(fairing.__file__)
```

Comment:
Non-blocker: are these steps necessary because the docker's python runtime doesn't have fairing installed? if so, my comment is that it is better to match exactly between the user's local python env and  the remote docker python env. We can discuss this in next issue/PR.

Reply:
Yes it’s to match the target runtime to fairing shim